### ISSUE TITLE
Should allow running as non-root

### ISSUE THREAD

**Issue by wernight** [paradoxxxzero/butterfly#119]
The Dockerized image should work as non-root user. For example by adding the docker flag `--user 1234`.

The main issue in the way is that if non-root (or if no `PASSWORD` is provided) it shouldn't run `chpass` in https://github.com/sekka1/butterfly/blob/master/docker/run.sh#L12

**Comment by sekka1** [comment]
Yeah, maybe it should just create the user.  However, if you run in non-root then that user can install anything via apt-get which requires root priv.  Then you have to start installing stuff like sudo.

**Comment by wernight** [comment]
I agree. That's why I'm not suggesting to make it the default, but to
support it.

For example my image is based on that Docker image and I install some apps
in my Dockerfile. Then I don't want to run as root when running it.

**Comment by wernight** [comment]
I tested a little more and found that currently it needs **an existing user** to work. This probably could be improved so it works even if the user doesn't exist (I think currently the user lookup in Python fails).

So the simplest right now to run as non-root is a `Dockerfile` like:

    FROM garland/butterfly
    # TODO: RUN apt-get install some-package if necessary.
    RUN useradd --create-home --uid 1234 --shell /bin/bash foo   # note: setting the shell is required
    USER 1234

    # Bypass the default entrypoint to avoid trying to call `chpasswd`.
    ENTRYPOINT ["butterfly.server.py"]
    CMD ["--unsecure", "--host=0.0.0.0"]
